tests/unit/test_client.py

```diff
@@ -3573,21 +3573,34 @@ def test_delete_table_w_not_found_ok_true(self):
 
         conn.api_request.assert_called_with(method="DELETE", path=path, timeout=None)
 
-    def _create_job_helper(self, job_config, client_method):
+    def _create_job_helper(self, job_config):
+        from google.cloud.bigquery import _helpers
+
         creds = _make_credentials()
         http = object()
         client = self._make_one(project=self.PROJECT, credentials=creds, _http=http)
 
-        client._connection = make_connection()
-        rf1 = mock.Mock()
-        get_config_patch = mock.patch(
-            "google.cloud.bigquery.job._JobConfig.from_api_repr", return_value=rf1,
-        )
-        load_patch = mock.patch(client_method, autospec=True)
+        RESOURCE = {
+            "jobReference": {"projectId": self.PROJECT, "jobId": mock.ANY},
+            "configuration": job_config,
+        }
+        conn = client._connection = make_connection(RESOURCE)
+        client.create_job(job_config=job_config)
 
-        with load_patch as client_method, get_config_patch:
-            client.create_job(job_config=job_config)
-        client_method.assert_called_once()
+        # if "copy" in job_config:
+        #     if "sourceTable" in job_config["copy"]:
+        #         source = _helpers._get_sub_prop(job_config, ["copy", "sourceTable"])
+        #         _helpers._del_sub_prop(job_config, ["copy", "sourceTable"])
+        #         _helpers._set_sub_prop(job_config, ["copy", "sourceTables"], [source])
+        if "query" in job_config:
+            _helpers._del_sub_prop(job_config, ["query", "destinationTable"])
```

Comment:
Do we still need to remove the destination table property from the config after my refactoring?

I thought we allow None to be passed to the job constructors for most properties now.

Reply:
I forget if we automatically remove the destination table to support query re-runs, but that seems like something we shouldn't remove by default, as explicit destination tables are also supported by the API.